File: Samples/GlobalizationPreferences/js/js/scenario3-region.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var outputText;

    var page = WinJS.UI.Pages.define("/html/scenario3-region.html", {
        ready: function (element, options) {
            document.getElementById("showResults").addEventListener("click", showResults);
            outputText = document.getElementById("outputText");
        }
    });

    function reportRegionData(region) {
        return  "Display Name: " + region.displayName + "\n" +
            "Native Name: " + region.nativeName + "\n" +
            "Currencies in use: " + region.currenciesInUse + "\n" +
            "Codes: " + region.codeTwoLetter + ", " + region.codeThreeLetter + ", " + region.codeThreeDigit + "\n\n";
    }

    function showResults() {
        // This scenario uses the Windows.Globalization.GeographicRegion class to
        // obtain the geographic region characteristics.
        var userGeoRegion = new Windows.Globalization.GeographicRegion();

        // This obtains the geographic region characteristics by providing a country or region code.
        var exampleGeoRegion = new Windows.Globalization.GeographicRegion("JP");

        // Display the results
        outputText.innerText = "User's Preferred Geographic Region\n" + reportRegionData(userGeoRegion) +
            "Example Geographic Region by region/country code (JP)\n" + reportRegionData(exampleGeoRegion);
    }
})();

File: Samples/GlobalizationPreferences/cppwinrt/Scenario3_Region.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario3_Region.h"
#include "Scenario3_Region.g.cpp"
#include "SampleConfiguration.h"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Globalization;
using namespace Windows::UI::Xaml;

namespace
{
    hstring ReportRegionData(GeographicRegion const& region)
    {
        return L"Display Name: " + region.DisplayName() + L"\n" +
            L"Native Name: " + region.NativeName() + L"\n" +
            L"Currencies in use: " + SDKTemplate::StringJoin(L", ", region.CurrenciesInUse()) + L"\n" +
            L"Codes: " + region.CodeTwoLetter() + L", " + region.CodeThreeLetter() + L", " + region.CodeThreeDigit() + L"\n\n";
    }

}
namespace winrt::SDKTemplate::implementation
{
    Scenario3_Region::Scenario3_Region()
    {
        InitializeComponent();
    }

    void Scenario3_Region::ShowResults_Click(IInspectable const&, RoutedEventArgs const&)
    {
        // This scenario uses the Windows.Globalization.GeographicRegion class to
        // obtain the geographic region characteristics.
        GeographicRegion userGeoRegion;

        // This obtains the geographic region characteristics by providing a country or region code.
        GeographicRegion exampleGeoRegion(L"JP");

        // Display the results
        OutputTextBlock().Text(L"User's Preferred Geographic Region\n" + ReportRegionData(userGeoRegion) +
            L"Example Geographic Region by region/country code (JP)\n" + ReportRegionData(exampleGeoRegion));
    }
}
